### PR TITLE
[SWPWA-409] Meta support - Default Meta Description is not taken into account 

### PR DESCRIPTION
**Pull request by dmitrijs-voronovs**
Now `title`, `description` and `keywords` meta are displayed correctly!

* Integrated `Meta` store
* Implemented `Meta` component
* Removed `react-helmet`
* Integrated into head `canonical url` and `title`

To test:
1. install [this](https://chrome.google.com/webstore/detail/ayima-page-insights/nbjdmeadccgbgmjhbahkgadffhpejgbd) extension
2. open any page and check extension for meta tags
3. in case it does not work, press `F12 > Elements` and check header for tags.

### CODE REVIEW

File: src/app/component/Meta/Meta.component.js [hunk 1]
```diff
@@ -19,53 +19,35 @@ import Helmet from 'react-helmet';
...
-            default_title = ''
+            canonical_url,
+            title_prefix,
+            title_suffix,
+            metadata,
+            title
         } = this.props;
 
         return (
             <Helmet
```
Comment (alfredsgenkins) [inline src/app/component/Meta/Meta.component.js]: Do you think we can get rid of Helmet? It seems unreasonable now. It weights like a 1/8 of a project.

Comment (alfredsgenkins) [reply src/app/component/Meta/Meta.component.js]: Let's just use portal from react-dom?

File: src/app/component/Meta/Meta.container.js [hunk 2]
```diff
@@ -1,13 +1,98 @@
...
+            title: this._getTitle(),
+            description: this._getDescription(),
+            keywords: this._getKeywords()
+        };
+
+        return this._generateMetaFromMetadata(meta);
+    }
+
+    _getMeta() {
+        return this._getMetadata();
```
Comment (alfredsgenkins) [inline src/app/component/Meta/Meta.container.js]: Why this method exists? :)

File: src/app/store/Meta/Meta.reducer.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+import {
+    UPDATE_META, UPDATE_META_FROM_COTEGORY, UPDATE_META_FROM_PRODUCT
```
Comment (alfredsgenkins) [inline src/app/store/Meta/Meta.reducer.js]: FYI prefer each on new line.

File: src/app/route/ProductPage/ProductPage.container.js [hunk 4]
```diff
@@ -79,10 +82,25 @@ export class ProductPageContainer extends PureComponent {
...
         if (pathname !== prevPathname) this._requestProduct();
+
+        if (id !== prevId) {
+            const dataSource = this._getDataSource();
+            const { updateMetaFromProduct } = this.props;
+
+            updateMetaFromProduct(dataSource);
+        }
+
         this._onProductUpdate();
```
Comment (alfredsgenkins) [inline src/app/route/ProductPage/ProductPage.container.js]: Why not to use on product update? Is it not reliable?

File: src/app/route/index.js [hunk 5]
```diff
@@ -261,11 +322,14 @@ class AppRouter extends PureComponent {
...
+                <Router history={ history }>
+                    { this.renderRouterContent() }
+                </Router>
+            </>
         );
     }
 }
 
-export default AppRouter;
+export default connect(mapStateToProps, mapDispatchToProps)(AppRouter);
```
Comment (alfredsgenkins) [inline src/app/route/index.js]: Maybe it is time to remove router from index.js? I think it is a bad place for it in the end of the day.

Comment (alfredsgenkins) [reply src/app/route/index.js]: That is FYI no action required.

File: src/app/store/Meta/Meta.reducer.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    return {
+        description: meta_description || description,
+        title: meta_title || name,
+        keywords: meta_keyword,
+        canonical_url
+    };
+};
+
+export const filterData = (data) => {
+    const updated = updateEveryTime.reduce((acc, key) => (
```
Comment (alfredsgenkins) [inline src/app/store/Meta/Meta.reducer.js]: Very complex construction. Please make more readable.

File: src/app/store/Meta/Meta.reducer.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+export const initialState = {
+    title: '',
+    title_prefix: '',
+    title_suffix: '',
+    description: '',
+    keywords: '',
+    canonical_url: ''
+};
+
+export const MetaReducer = (state = initialState, action) => {
```
Comment (alfredsgenkins) [inline src/app/store/Meta/Meta.reducer.js]: Use dispatcher and single action, as all your actions in reality do the same thing. See how breadcrumbs are done :)

File: src/app/route/ProductPage/ProductPage.container.js (hunk 4 above)
Comment (dmitrijs-voronovs) [reply src/app/route/ProductPage/ProductPage.container.js]: I went to magento admin to see if it is possible to set defferent meta for each configuration of configurable product and it turned out that it is not. 
Also went to SEO specialist for the same purpose. Was told to get single meta for configurable product.
Therefore it is necessary to update meta only on product change, what I did.

File: src/app/route/index.js (hunk 5 above)
Comment (dmitrijs-voronovs) [reply src/app/route/index.js]: Yes!

File: src/app/component/Meta/Meta.component.js (hunk 1 above)
Comment (dmitrijs-voronovs) [reply src/app/component/Meta/Meta.component.js]: Ok! I will explore it, make relevant changes and tell you when it's done!

Comment (dmitrijs-voronovs) [reply src/app/component/Meta/Meta.component.js]: It works!
I removed hardcoded title in html index templates and used portals to render
* title
* canonical url (where relevant)
* meta tags

**Pull request merged**